crates/ruff_server/src/session/workspace/configuration.rs

```diff
@@ -0,0 +1,74 @@
+use anyhow::anyhow;
+use lsp_types::Url;
+use ruff_workspace::resolver::{ConfigurationTransformer, Relativity};
+use std::{
+    collections::BTreeMap,
+    path::{Path, PathBuf},
+    sync::Arc,
+};
+
+#[derive(Default)]
+pub(crate) struct RuffConfiguration {
+    // settings to pass into the ruff linter
+    pub(crate) linter: ruff_linter::settings::LinterSettings,
+    // settings to pass into the ruff formatter
+    pub(crate) formatter: ruff_workspace::FormatterSettings,
+}
```

Comment:
The naming here becomes slightly confusing. I can see why you would name it `Configuration` but we generally  use the term `Settings` for deserialized, merged and resolved "ruff configurations" and only use the term `Configuration` for deserialized configuration files (but not resolved).